tests/model/test_model.py

```diff
@@ -337,6 +337,14 @@ def test_statements():
     assign = Assignment.create(Expr.function('A', Expr.symbol('t')), Expr.symbol('t'))
     Model.create('model', statements=Statements() + assign)
 
+    assign = Assignment.create(Expr.function('A', 0), Expr.symbol('t'))
+    with pytest.raises(ValueError, match='Symbol t is not defined'):
```

Comment:
Why shouldn't `t` be defined here?

Reply:
Because `t` is not defined in `A(0) = t`? Or should we always allow `t`?

Reply:
We should always allow `t` when the model has an ODE system. Perhaps `t` is the default and can be changed I don't remember.

Reply:
Yes, but if the model has no ODE system then it should raise if `t` is not defined, right? I think in the previous implementation we would always allow `t`.

Reply:
Yes, that is what we want.